site/src/xServices/users/usersXService.ts

```diff
@@ -70,393 +80,455 @@ export type UsersEvent =
       userId: TypesGen.User["id"]
       roles: TypesGen.Role["name"][]
     }
+  // Filter
+  | { type: "UPDATE_FILTER"; query: string }
+  // Pagination
+  | { type: "UPDATE_PAGE"; page: string }
 
-export const usersMachine = createMachine(
```

Comment:
A lot of this diff is just because I used the visualizer and it reformatted, so I'll comment where it's relevant